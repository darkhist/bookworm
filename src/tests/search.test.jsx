import React from 'react';
import { shallow, mount } from 'enzyme';

import Search from '../components/search/search';

describe('Search', () => {
  const fn = jest.fn();
  const valid = true;

  it('should render without crashing', () => {
    const search = shallow(<Search onChange={fn} valid={valid} />);
    expect(search).toMatchSnapshot();
  });

  it('should have the correct init state', () => {
    const search = shallow(<Search onChange={fn} valid={valid} />);
    expect(search.state('term')).toEqual('');
  });

  it('should have an input', () => {
    const search = mount(<Search onChange={fn} valid={valid} />);
    const input = search.find('input');
    expect(input.exists()).toBeTruthy();
  });

  it('should update the state on input change', () => {
    const search = mount(<Search onChange={fn} valid={valid} />);
    const input = search.find('input');

    input.simulate('change', {
      target: {
        value: 'Harry Potter'
      }
    });
    expect(search.state('term')).toEqual('Harry Potter');

    input.simulate('change', {
      target: {
        value: 'The Great Gatsby'
      }
    });
    expect(search.state('term')).toEqual('The Great Gatsby');
  });
});
