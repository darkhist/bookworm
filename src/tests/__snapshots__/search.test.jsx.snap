// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render without crashing 1`] = `
<styled.div>
  <styled.input
    aria-label="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <styled.span
    valid={true}
  >
    Searches cannot be empty, and the following characters are not allowed: #, %, &
  </styled.span>
</styled.div>
`;
